### ISSUE TITLE
Drush dependency throwing Fatal Php Error

### ISSUE THREAD

**Issue by tbutterf** [drush-ops/drush#5995]
**Describe the bug**
Running `composer why webflo/drupal-finder` indicates that it is required by drush/drush. A recent commit made to drupal-finder is breaking Drush on Drupal 10 sites.

**To Reproduce**
What did you do?
I ran `composer update` which brought in the latest commit on webflo/drupal-finder. Then I ran `drush cr` to rebuild the cache.

**Expected behavior**
What did you expect would happen?
I expected it to rebuild the cache successfully.

**Actual behavior**
What happened instead?

I got the following error:
```
Fatal error: Uncaught TypeError: is_link(): Argument #1 ($filename) must be of type string, Drush\Config\Environment given in /var/www/html/vendor/webflo/drupal-finder/src/DrupalFinder.php:153
Stack trace:
#0 /var/www/html/vendor/webflo/drupal-finder/src/DrupalFinder.php(153): is_link(Object(Drush\Config\Environment))
#1 /var/www/html/vendor/webflo/drupal-finder/src/DrupalFinder.php(75): DrupalFinder\DrupalFinder->discoverRoots(Object(Drush\Config\Environment))
#2 /var/www/html/vendor/drush/drush/src/Preflight/Preflight.php(310): DrupalFinder\DrupalFinder->__construct(Object(Drush\Config\Environment))
#3 /var/www/html/vendor/drush/drush/src/Runtime/Runtime.php(58): Drush\Preflight\Preflight->preflight(Array)
#4 /var/www/html/vendor/drush/drush/src/Runtime/Runtime.php(40): Drush\Runtime\Runtime->doRun(Array, Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 /var/www/html/vendor/drush/drush/drush.php(139): Drush\Runtime\Runtime->run(Array)
#6 /var/www/html/vendor/drush/drush/drush(4): require('/var/www/html/v...')
#7 /var/www/html/vendor/bin/drush(119): include('/var/www/html/v...')
#8 {main}
  thrown in /var/www/html/vendor/webflo/drupal-finder/src/DrupalFinder.php on line 153
```

**Workaround**
Is there another way to do the desired action?
For now, I had to roll back to webflo/drupal-finder version 1.2.2, which does not cause the fatal error.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 12.4.3
| Drupal version? | 10.2.5
| PHP version     | 8.1.27
| OS?             | Mac

**Additional information**
I have [opened an issue ticket](https://github.com/webflo/drupal-finder/issues/67) on that repository.

**Comment by weitzman** [comment]
As a starting point, I suggest upgrading to latest supported Drush. if it still happens, we can dig further.

**Comment by tbutterf** [comment]
@weitzman That seems to have solved it. I guess it's a false alarm. Thanks!

**Comment by tbutterf** [comment]
@weitzman, unfortunately, after further inspection, upgrading to the latest Drush version only solves it for sites using Drush 12.x on Drupal 10. For sites using Drush 11.x on Drupal 9, it still breaks Drush. The only solution I have found to get it to work is to downgrade webflo/drupal-finder to 1.2.2. But, it seems like maybe a change to the Drush 11.x branch similar to whatever happened in the 12.x branch might solve this. Otherwise, this must be fixed in the webflo product itself.

**Comment by weitzman** [comment]
Drush 11 is EOL since Nov 2023.

**Comment by webflo** [comment]
@tbutterf Could you provide a stack strace for Drupal 9 and Drush 11.x?

**Comment by tbutterf** [comment]
@webflo and @weitzman. Apologies. I was getting some errors in my CI builds yesterday, and I didn't look closely enough at the errors and assumed it was caused by the same thing. Today when I tried to replicate the issue locally, and in CI, I realized that the problem is not occurring in Drupal 9 with Drush 11.x. 🤦 The error I was getting was unrelated. So, it looks like Drush 11 with Drupal 9 is okay with the latest update. My mistake. Thanks for your quick responses!
